File: 769.py
class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        res = 0
        maxVal = 0
        for i, x in enumerate(arr):
            maxVal = max(maxVal, x)
            if i == maxVal:
                res += 1
        return res

File: 769.cpp
struct Solution {
    inline int maxChunksToSorted(vector<int>& arr) {
        int res = 0, maxVal = 0;
        for (int i = 0; i < arr.size(); ++i){
            maxVal = max(maxVal, arr[i]);
            if (i == maxVal){
                ++res;
            }
        }
        return res;
    }
};
